### DISCUSSION TITLE
Problem starting on windows shared hosting with Plesk

### DISCUSSION THREAD

**Discussion by stramaxxa** [directus/directus#5106] [Archive]
I have windows shared hosting  with plesk and I had several difficulties getting Directus to run.
Node.js Version: 12.19.1  

**First:**
Plesk to run a node.js app requires a js file in the site root.
I created the start.js file in the root:
```
const d = require("./node_modules/directus/dist/start.js");
d.default();
```

**Second**
I wrote a routing rule in the web.config:
```
<system.webServer>
	<rewrite>
		<clear/>
		<rules>
			<rule name="DynamicContent">
				<action type="Rewrite" url="start.js" />
			</rule>
		</rules>
	</rewrite>	
</system.webServer>
```

**Third**
I edited the _node_modules/directus/dist/start.js_ file at line 46 to define the default port
directus default port doesn't work
```
const port = process.env.PORT;
```

**Fourth**
I had problems in the _node_modules/directus/dist/middleware/authenticate.js_ file on line 50
"_Cannot read property 'startsWith' of undefined_" I don't know why "req.ip" is undefined.
i forced replacing the line
```
   ip: req.ip.startsWith('::ffff:') ? req.ip.substring(7) : req.ip,
```
with this:
```
  ip: req.ip,
```
seems to work, but on login the problem of the "null" ip still recurs :
```
insert into `directus_activity` (`action`, `collection`, `ip`, `item`, `user`, `user_agent`) values (.....) - ER_NO_DEFAULT_FOR_FIELD: Field 'ip' doesn't have a default value 
Error: ER_NO_DEFAULT_FOR_FIELD: Field 'ip' doesn't have a default value
```
I then added a default value for the "id" field in the "directus_activity" table and now everything seems to work.

**Comment by rijkvanzanten** [comment]
I would highly recommend looking into why Plesk prevents the IP address from being readable, instead of disabling it like this in Directus itself 🤔 Things like IP-address based role access will break completely by removing the availability of the ip address

**Reply by stramaxxa** [reply]
Thank you for adding the installation guide on Shared Hosting with Plesk ([here](https://docs.directus.io/guides/installation/plesk/)).

The guide works correctly and Directius installs without problems.

However, I had to add some code to the web.config to resolve a "405 Method Not Allowed" error when deleting the data row.

`<system.webServer>
      <modules>
	  	<remove name="WebDAVModule" />
      </modules>
      <handlers>
	  	<remove name="WebDAV" />
    </handlers>
<system.webServer>`

**Reply by rijkvanzanten** [reply]
@stramaxxa Wanna add that to the docs in a PR? 🙂
